src/components/notes/TrashedBanner/TrashedBanner.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+
+import Alert from 'cozy-ui/transpiled/react/Alert'
+import Buttons from 'cozy-ui/transpiled/react/Buttons'
+import TrashDuotoneIcon from 'cozy-ui/transpiled/react/Icons/TrashDuotone'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import { useClient } from 'cozy-client'
+import { useAlert } from 'cozy-ui/transpiled/react/providers/Alert'
+import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
+
+import { DestroyConfirm } from 'components/notes/TrashedBanner/DestroyConfirm'
+
+const TrashedBanner = ({ noteId, isPublic, returnUrl }) => {
+  const { t } = useI18n()
+  const client = useClient()
+  const { showAlert } = useAlert()
+  const { isMobile } = useBreakpoints()
+  const [
+    isDestroyConfirmationDisplayed,
+    setDestroyConfirmationDisplayed
+  ] = useState(false)
+
+  const [isBusy, setBusy] = useState(false)
+
+  const restore = async () => {
+    try {
+      await client.collection('io.cozy.files').restore(noteId)
+      showAlert(t('TrashedBanner.restoreSuccess'), 'secondary')
+    } catch {
+      showAlert(t('TrashedBanner.restoreError'), 'error')
+    } finally {
+      setBusy(false)
+    }
+  }
```

Comment:
We can't use `useMutation` because of the need to call a specific method of the collection (restore)?

Reply:
Unfortunately, the useMutation API only allows the use of a save and not to pass a function as in its react-query equivalent. Perhaps this is something that needs to be rethink. We could transform the current one into `useSave` and keep `useMutation` for more generic cases like this one